ui/pages/smart-transactions/smart-transaction-status-page/smart-transactions-status-page.test.js

```diff
@@ -0,0 +1,138 @@
+    const mockStore = createSwapsMockStore();
+    const latestSmartTransaction =
+      mockStore.metamask.smartTransactionsState.smartTransactions[
+        CHAIN_IDS.MAINNET
+      ][1];
+    latestSmartTransaction.status = SmartTransactionStatuses.REVERTED;
+    requestState.smartTransaction = latestSmartTransaction;
+    const store = configureMockStore(middleware)(mockStore);
+    const { getByText } = renderWithProvider(
+      <SmartTransactionStatusPage requestState={requestState} />,
+      store,
+    );
+    expect(getByText('Your transaction failed')).toBeInTheDocument();
+    expect(
+      getByText(
+        'Sudden market changes can cause failures. If the problem continues, reach out to MetaMask customer support.',
+      ),
+    ).toBeInTheDocument();
+  });
+
+  it('renders the "cancelled" STX status', () => {
+    const mockStore = createSwapsMockStore();
+    const latestSmartTransaction =
+      mockStore.metamask.smartTransactionsState.smartTransactions[
+        CHAIN_IDS.MAINNET
+      ][1];
+    requestState.smartTransaction = latestSmartTransaction;
+    latestSmartTransaction.status = SmartTransactionStatuses.CANCELLED;
+    const store = configureMockStore(middleware)(mockStore);
+    const { getByText } = renderWithProvider(
+      <SmartTransactionStatusPage requestState={requestState} />,
+      store,
+    );
+    expect(getByText('Your transaction was canceled')).toBeInTheDocument();
+    expect(
+      getByText(
+        `Your transaction couldn't be completed, so it was canceled to save you from paying unnecessary gas fees.`,
+      ),
+    ).toBeInTheDocument();
+  });
+
+  it('renders the "deadline_missed" STX status', () => {
+    const mockStore = createSwapsMockStore();
+    const latestSmartTransaction =
+      mockStore.metamask.smartTransactionsState.smartTransactions[
+        CHAIN_IDS.MAINNET
+      ][1];
+    latestSmartTransaction.status =
+      SmartTransactionStatuses.CANCELLED_DEADLINE_MISSED;
+    requestState.smartTransaction = latestSmartTransaction;
+    const store = configureMockStore(middleware)(mockStore);
+    const { getByText } = renderWithProvider(
+      <SmartTransactionStatusPage requestState={requestState} />,
+      store,
+    );
+    expect(getByText('Your transaction was canceled')).toBeInTheDocument();
+  });
+
+  it('renders the "unknown" STX status', () => {
+    const mockStore = createSwapsMockStore();
+    const latestSmartTransaction =
+      mockStore.metamask.smartTransactionsState.smartTransactions[
+        CHAIN_IDS.MAINNET
+      ][1];
+    latestSmartTransaction.status = SmartTransactionStatuses.UNKNOWN;
+    requestState.smartTransaction = latestSmartTransaction;
+    const store = configureMockStore(middleware)(mockStore);
+    const { getByText } = renderWithProvider(
+      <SmartTransactionStatusPage requestState={requestState} />,
+      store,
+    );
+    expect(getByText('Your transaction failed')).toBeInTheDocument();
```

Comment:
Depending on the complexity of the component, we could also benefit from snapshot tests so we'd be notified if any changes update the visual.